_studio/shared/src/libmfx_core_vaapi.cpp

```diff
@@ -1243,59 +1243,70 @@ void VAAPIVideoCORE::ReleaseHandle()
-            return MFX_WRN_PARTIAL_ACCELERATION;
-        break;
-    case MFX_CODEC_JPEG:
-        if (par->mfx.FrameInfo.Width > 8192 || par->mfx.FrameInfo.Height > 8192)
-            return MFX_WRN_PARTIAL_ACCELERATION;
-        break;
-    case MFX_CODEC_VP8:
-        if (m_HWType < MFX_HW_BDW)
-            return MFX_ERR_UNSUPPORTED;
-    case MFX_CODEC_VP9:
-        break;
-    default:
-        return MFX_ERR_UNSUPPORTED;
-    }
+    MFX_CHECK(m_Display, MFX_ERR_DEVICE_FAILED);
 
-    if (MFX_HW_SNB == m_HWType)
-    {
-        if (par->mfx.FrameInfo.Width > 1920 || par->mfx.FrameInfo.Height > 1200)
-            return MFX_WRN_PARTIAL_ACCELERATION;
-    }
-    else
+    VaGuidMapper mapper(guid);
+    VAProfile req_profile = mapper.profile;
+    VAEntrypoint req_entrypoint = mapper.entrypoint;
+    mfxI32 va_max_num_entrypoints = vaMaxNumEntrypoints(m_Display);
+    mfxI32 va_max_num_profiles = vaMaxNumProfiles(m_Display);
+    MFX_CHECK(va_max_num_entrypoints && va_max_num_profiles, MFX_ERR_UNSUPPORTED);
+
+    //driver always support VAProfileNone
+    if (req_profile != VAProfileNone)
     {
-        if (MFX_CODEC_JPEG != par->mfx.CodecId &&
-            MFX_CODEC_HEVC != par->mfx.CodecId &&
-        (par->mfx.FrameInfo.Width > 4096 || par->mfx.FrameInfo.Height > 4096))
-            return MFX_WRN_PARTIAL_ACCELERATION;
+        vector <VAProfile> va_profiles (va_max_num_profiles, VAProfileNone);
+
+        //ask from driver about profile support
+        VAStatus va_sts = vaQueryConfigProfiles(m_Display,
+                            va_profiles.data(), &va_max_num_profiles);
+        MFX_CHECK(va_sts == UMC_OK, MFX_ERR_UNSUPPORTED);
+
+        //check profile support
+        auto it_profile = find(va_profiles.begin(), va_profiles.end(), req_profile);
+        MFX_CHECK(it_profile != va_profiles.end(), MFX_ERR_UNSUPPORTED);
     }
 
+    vector <VAEntrypoint> va_entrypoints (va_max_num_entrypoints, static_cast<VAEntrypoint> (0));
+
+    //ask from driver about entrypoint support
+    VAStatus va_sts = vaQueryConfigEntrypoints(m_Display, req_profile,
+                    va_entrypoints.data(), &va_max_num_entrypoints);
+    MFX_CHECK(va_sts == UMC_OK, MFX_ERR_UNSUPPORTED);
+
+    //check entrypoint support
+    auto it_entrypoint = find(va_entrypoints.begin(), va_entrypoints.end(), req_entrypoint);
+    MFX_CHECK(it_entrypoint != va_entrypoints.end(), MFX_ERR_UNSUPPORTED);
+
+    VAConfigAttrib attr[] = {{VAConfigAttribMaxPictureWidth,  0},
+                             {VAConfigAttribMaxPictureHeight, 0}};
+
+    //ask from driver about support
+    va_sts = vaGetConfigAttributes(m_Display, mapper.profile,
+                                   mapper.entrypoint,
+                                   attr, sizeof(attr)/sizeof(*attr));
+
+    MFX_CHECK(va_sts == UMC_OK, MFX_ERR_UNSUPPORTED);
+
+    //check resolution video
+    MFX_CHECK(attr[0].value != VA_ATTRIB_NOT_SUPPORTED, MFX_ERR_UNSUPPORTED);
```

Comment:
there won't be any harm from my point of view if you combine conditions for same fields and reduce codelines, i.e.

MFX_CHECK(attr[0].value != VA_ATTRIB_NOT_SUPPORTED && attr[0].value != 0, MFX_ERR_UNSUPPORTED);

But it's completely up to you whether to change it or not

Reply:
I like author's variant better...